extensions/machine-learning-services/src/common/httpClient.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as request from 'request';
+import * as constants from './constants';
+
+const DownloadTimeout = 20000;
+export class HttpClient {
+	public download(downloadUrl: string, targetPath: string, backgroundOperation: azdata.BackgroundOperation, outputChannel: vscode.OutputChannel): Promise<void> {
+		return new Promise((resolve, reject) => {
+
+			let totalMegaBytes: number | undefined = undefined;
+			let receivedBytes = 0;
+			let printThreshold = 0.1;
+			let downloadRequest = request.get(downloadUrl, { timeout: DownloadTimeout })
```

Comment:
no get is not async but data is returned with events so that's why its' wrapped by promise

---
In reply to: [366615481](https://github.com/microsoft/azuredatastudio/pull/8870#discussion_r366615481) [](ancestors = 366615481)